pkg/cloud/services/eks/iam/iam.go

```diff
@@ -461,7 +462,7 @@ func (s *IAMService) FindAndVerifyOIDCProvider(cluster *eks.Cluster) (string, er
 			return "", errors.Wrap(err, "error getting provider")
 		}
 		// URL should always contain `https`.
-		if *provider.Url != issuerURL.String() {
+		if *provider.Url != issuerURL.String() && *provider.Url != strings.Replace(issuerURL.String(), "https://", "", 1) {
```

Comment:
I think this check could be enhanced, may be like check if the err is OIDCAlreadyExist, then break, else have this https check. wdyt?

Reply:
Hi @Ankitasw, first of all thanks for reviewing the PR, it seems like a good idea, I'll get to work :)

Reply:
Hi @Ankitasw
I'm sorry, it has been impossible for me to dedicate time to it, this week it will be closed

Reply:
Hi @Ankitasw,
I have been reviewing the issue, function FindAndVerifyOIDCProvider never return "https" in the identity provider response so never complain the condition and always returns "" and because of that, the function CreateOIDCProvider always return the error "EntityAlreadyExists" if OIDC already exists; if you agree we could enhance the check just using this code (if we are sure https is always present) like this:
if *provider.Url != strings.Replace(issuerURL.String(), "https://", "", 1) {
            continue
        }

Reply:
There is a requirement in the docs stating:
> The URL must begin with https://.

So I think that's a safe bet.